id.go

```diff
@@ -114,7 +115,9 @@ func readMachineID() []byte {
 	id := make([]byte, 3)
 	if hostname, err := os.Hostname(); err == nil {
 		hw := md5.New()
-		hw.Write([]byte(hostname))
+		if _, errW := hw.Write([]byte(hostname)); errW != nil {
+			panic(fmt.Errorf("xid: cannot write hostname hash: %v;", errW))
```

Comment:
This error can never happen.

Reply:
if the code isn't changing, it won't, I totally agree.
I just have the habit of never silencing errors, just in case the future, the code changes above this line and suddenly it starts crashing randomly without apparent reason.
but I totally agree it's very unlikely to happen.

Up to you if you want me to remove that.

Reply:
By nature, a md5 written into a memory address is just about manipulating registers and memory and so will never fail (or if it does, it will panic earlier). I agree with the no silence error stance, but I also value small code bases.

I'm ok to accept this change. Please reuse `err` so we don't have `errW`.